ompi/attribute/attribute_predefined.c

```diff
@@ -165,14 +170,23 @@ int ompi_attr_create_predefined(void)
 
     /* If the universe size is set, then use it. Otherwise default
      * to the size of MPI_COMM_WORLD */
-    univ_size = getenv("OMPI_UNIVERSE_SIZE");
-    if (NULL == univ_size || (usize = strtol(univ_size, NULL, 0)) <= 0) {
-        ret = set_f(MPI_UNIVERSE_SIZE, ompi_comm_size(MPI_COMM_WORLD));
+    opal_process_name_t wildcard = {OMPI_PROC_MY_NAME->jobid, OPAL_VPID_WILDCARD};
+    OPAL_MODEX_RECV_VALUE_OPTIONAL(rc, OPAL_PMIX_UNIV_SIZE, &wildcard, &usizeptr, OPAL_UINT32);
+    if ( 0 == rc && usize != 0) {
+       ret = set_f(MPI_UNIVERSE_SIZE, usize);
     } else {
-        ret = set_f(MPI_UNIVERSE_SIZE, usize);
+       univ_size = getenv("OMPI_UNIVERSE_SIZE");
+       if (NULL == univ_size || (strtol_result = strtol(univ_size, NULL, 0)) <= 0) {
+           ret = set_f(MPI_UNIVERSE_SIZE, ompi_comm_size(MPI_COMM_WORLD));
+       } else {
+           /* PMIx only supports sizes up to unsigned int and so will 
+            * environment variable approach */
+           usize = (unsigned int) strtol_result;
+           ret = set_f(MPI_UNIVERSE_SIZE, usize);
+       }
     }
     if (OMPI_SUCCESS != ret) {
-        return ret;
+         return ret;
```

Comment:
I would suggest removing the extra space, but it looks like PR #9288 went in with the space. So probably best to match them at this point.